Library/Homebrew/cask/lib/hbc/system_command.rb

```diff
@@ -82,10 +93,11 @@ def expanded_command
     end
 
     def each_output_line(&b)
+      popen3_env = sudo? ? env : {}
       executable, *args = expanded_command
 
       raw_stdin, raw_stdout, raw_stderr, raw_wait_thr =
-        Open3.popen3({ "PATH" => path }, [executable, executable], *args, **options)
+        Open3.popen3(popen3_env, [executable, executable], *args, **options)
```

Comment:
Does it hurt to always pass `env` regardless of `sudo`?

Reply:
I’m torn on this one. Passing `env` unconditionally probably won’t hurt; that said, I feel it would be confusing from a maintenance point of view because 1. it has no actual effect (at least with the three variables we’re introducing here); more importantly, 2. in the `sudo` case, it may obscure the presence of the real assignments, causing confusion; finally, 3. at some point, the redundant assignment may cause a future maintainer to mistakenly remove the important part. (The latter is not really unlikely given that admittedly, I haven’t made it obvious from the code why the special treatment is there after all.)

I’ll gladly change it but I’d have to add inline comments in order to mitigate 2. and 3.  
What do you think?